### ISSUE TITLE
TF 0.13 bug with The "count" value depends on resource attributes that cannot be determined until apply

### ISSUE THREAD

**Issue by bharathkkb** [hashicorp/terraform#26579]
### Terraform Version

```
0.13.4
```

### Terraform Configuration Files

- Apply this config which will fail due to ` machine_type = "foo"`
```terraform
provider "google" {
  version = "~> 3.42.0"
  project  = var.project
}

variable "project" {
    type = string
}

resource "google_project_service" "container_api" {
  service                    = "container.googleapis.com"
  disable_on_destroy         = false
  disable_dependent_services = false
}

resource "google_container_cluster" "primary" {
  name     = "my-gke-cluster-1"
  location = "us-central1-a"
  remove_default_node_pool = true
  initial_node_count       = 1
  depends_on = [google_project_service.container_api]
}

resource "google_container_node_pool" "pools" {
  count = 2
  name       = "my-node-pool"
  location   = "us-central1-a"
  cluster    = google_container_cluster.primary.name
  node_count = 1
  node_config {
    machine_type = "foo"
  }
}

resource "random_pet" "pets" {
count = length([for pool in google_container_node_pool.pools : pool.name])
}
```
- Subsequent `terraform apply` fails with
```
The "count" value depends on resource attributes that cannot be determined
until apply, so Terraform cannot predict how many instances will be created.
To work around this, use the -target argument to first apply only the
resources that the count depends on.
```
### Crash Output
N/A

### Expected Behavior
Plan should be successful as it is in 0.12.29

### Actual Behavior
```
The "count" value depends on resource attributes that cannot be determined
until apply, so Terraform cannot predict how many instances will be created.
To work around this, use the -target argument to first apply only the
resources that the count depends on.
```

### Steps to Reproduce
Listed above

### Additional Context

For `0.13.4` using 
```
terraform apply -refresh=false 
```
which bypasses the initial refresh seems to be a workaround.

### References

x-ref: https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/issues/690
Might be related to 
- https://github.com/hashicorp/terraform/issues/26078

**Comment by Gael101** [comment]
I do have the same behaviour when running "terraform destroy"on this resource
`resource "aws_efs_mount_target" "efs_target" {
    count = local.subnet_count
    file_system_id = aws_efs_file_system.efs.id
    subnet_id      = element(data.terraform_remote_state.vpc_state.outputs.private_subnet_id, count.index)
    security_groups = [data.terraform_remote_state.sg_state.outputs.sg_back_web_id]
}`
`subnet_count="${length(data.terraform_remote_state.vpc_state.outputs.private_subnet_id)}"`
From what i'm seeing in the console , the mount targets were destroyed during the destroy of the below resource
`resource "aws_efs_file_system" "efs" {
    creation_token = "${local.project_name}-${local.workspace}-efs"
    tags = {
        project_name   = local.project_name
        env            = local.workspace
    }
}`
But this should not rise an error

**Comment by jbardin** [comment]
Thanks @bharathkkb 

The fact that `-refresh=false` confirms that this is something resolved by the changes in 0.14. There will no longer be a separate refresh phase internally, and this type of skew between configuration and state should no longer be a factor. 

Thanks!
